src/lib/trivy/trivy-collector.ts

```diff
@@ -21,8 +21,14 @@ export class TrivyCollector extends AnalysisCollectorBase {
     }
 
     const args = ['--quiet', 'image', '--security-checks', 'vuln,config', '--exit-code', '0', '-f', 'json', '--light', imageName];
-    const output = await this.spawn('trivy', args, options);
 
+    let output;
+    try {
+      output = await this.spawn('trivy', args, options);
+    }
+    catch (e: unknown) {
```

Comment:
I don't think `unknown` is the appropriate type here. `catch` will always catch `Error` or a subtype. The `${ e as string }` bit is just masking implicit behaviour in the javascript to cast (`.toString()`) the error because it's inside a string interpolation block.

Reply:
Our linter doesn't allow us to do `catch(e)`. Nor `catch (e: Error)`. Nor `catch (e: any)`. This is the [linter's complaint](https://typescript.tv/best-practices/error-ts1196-catch-clause-variable-type-annotation/)

Is there a way to use anything else (besides `unknown`)?

Reply:
I've seen how the other check like zap or checkov implement the `spawn` function and they just let the process terminate with the raw exception coming from `analysis-collector-base.ts` if there are any errors. I'm not really sure if we need to add another layer to manipulate the error message.

Reply:
> Is there a way to use anything else (besides unknown)?

I don't think it's worth doing the type assertions inside the catch block. It's fine as-is.